engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/runtime/GetStackTraceNode.java

```diff
@@ -23,34 +28,83 @@ public static GetStackTraceNode create() {
     return new GetStackTraceNode();
   }
 
+  @CompilerDirectives.TruffleBoundary
+  private static EnsoObject wrapStackTraceElements(List<TruffleStackTraceElement> elements) {
+    var arr = new Object[elements.size()];
+    for (var i = 0; i < arr.length; i++) {
+      var e = elements.get(i);
+      arr[i] = e.getGuestObject();
+    }
+    var vector = ArrayLikeHelpers.asVectorWithCheckAt(arr);
+    try {
+      return filterStackTraceVector(InteropLibrary.getUncached(), vector);
+    } catch (UnsupportedMessageException ex) {
+      return ArrayLikeHelpers.empty();
+    }
+  }
+
+  private static boolean includeFrame(InteropLibrary iop, Object elem)
+      throws UnsupportedMessageException {
+    if (!iop.hasSourceLocation(elem)) {
+      return false;
+    }
+    var ss = iop.getSourceLocation(elem);
+    if (ss == null) {
+      return false;
+    }
+    var src = ss.getSource();
+    return src != null && src.hasCharacters() && !src.isInternal();
+  }
+
+  private static EnsoObject filterStackTraceVector(InteropLibrary iop, Object elements)
+      throws UnsupportedMessageException {
+    var size = iop.getArraySize(elements);
+    var count = 0;
+    for (long i = 0; i < size; i++) {
+      try {
+        var element = iop.readArrayElement(elements, i);
+        if (includeFrame(iop, element)) {
+          count++;
+        }
+      } catch (InvalidArrayIndexException ex) {
```

Comment:
Instead of a complete empty catch body, what about at least throwing `throw new AssertionError(ex);`? To convey that this is an unexpected exception.

Reply:
9d4c333fe4 will report unexpected exceptions when `-ea`